docs/DeveloperGuide.md

```diff
@@ -2,37 +2,220 @@
+
+> **Example**:\
+> `addIncome money /amt 1000 /date 2021-10-10` with prefixes array `{"addIncome", "/amt", "/date"}` will be tokenized into a hash map with the following key-value pairs:\
+> `addIncome: money`\
+> `amt: 1000`\
+> `date: 2021-10-10`
+
+Delimiters can also be specified as optional by adding ":optional" to the end. This allows the user to omit the prefix when entering the command.
+
+#### Implementation
+
+The tokenize function is implemented by the StringTokenizer class. It takes in the user input and the prefixes array and returns a hashmap of the user input.
+
+![](./images/stringTokenizer.png)
+
+**Step 1**\
+Check if all necessary prefixes are present in the user input. If not, throw an exception.
+
+**Step 2**\
+Generate list of PrefixWithPosition (containing the prefix/delimiter and the position in the string) for the delimiters in the user input
+
+**Step 3**\
+Sort the list of PrefixWithPosition by position in order to get the prefixes in the correct order
+
+**Step 4**\
+Create a hashmap of the user input by splitting the string using the prefixes' positions and adding the key-value pairs to the hashmap
+
+### Transaction Filter
+The filter function assists in filtering both expenses and incomes according to the specified criterion from user's input.
+
+#### Implementation
+![](./images/FilterTransaction.png)
+
+**Step 1**\
+The filterBy() method is called, returning a FindParser object
+
+**Step 2**\
+A FilterTransaction object is constructed, with the FindParser object as one of its parameters
+
+**Step 3**\
+From the expenseStatement and incomeStatement, the financialStatement is constructed. 
+
+**Step 4**\
+Execution of the FilterTransaction class calls method findTransaction(), which prints expenses and incomes that match the provided criterion
+
+### File Storage
+
+The FileStorage class in the CashLeh? application facilitates the reading and writing of financial transaction data to and from a file. It simplifies the process of managing user data, making it easily accessible and modifiable through the following methods:
+
+#### Reading Data
+![](./images/readFromFile.png)
+
+**Step 1**\
+The `readFromFile` method checks if the specified file exists. If not, it creates an empty file with the user's name as the file path
+
+**Step 2**\
+After ensuring the file exists, the method parses the data by reading each line from the file. It distinguishes between income and expense transactions based on a type identifier
+
+**Step 3**\
+For each transaction line, the method validates the format. It checks if essential details like transaction type, description, amount, and date are present. If any information is missing or the format is corrupted, a `CashLehFileCorruptedException` is thrown
+
+**Step 4**\
+Valid transactions are used to create `Income` or `Expense` objects, depending on their type. These objects are added to the provided `IncomeStatement` and `ExpenseStatement` to populate the financial data.
```

Comment:
In general, the explanations of sequence diagrams throughout your DG are well written and easy to follow.